Sources/Fog/FogSyncChecker.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  Copyright (c) 2020-2021 MobileCoin. All rights reserved.
+//
+
+import Foundation
+
+protocol FogSyncCheckable {
+    var viewsHighestKnownBlock: UInt64 { get }
+    var ledgersHighestKnownBlock: UInt64 { get }
+    var consensusHighestKnownBlock: UInt64 { get }
+    var currentBlockIndex: UInt64 { get }
+    var fogSyncThreshold: UInt64 { get }
+    
+    func inSync() -> Result<(), FogSyncError>
+    func setLedgersHighestKnownBlock(_:UInt64)
+    func setViewsHighestKnownBlock(_:UInt64)
+    func setConsensusHighestKnownBlock(_:UInt64)
+}
+
+extension FogSyncCheckable {
+    func inSync() -> Result<(), FogSyncError> {
+        guard viewLedgerOutOfSync == false else {
+            return .failure(.viewLedgerOutOfSync(viewsHighestKnownBlock, ledgersHighestKnownBlock))
+        }
+        guard consensusOutOfSync == false else {
+            return .failure(.consensusOutOfSync(consensusHighestKnownBlock, currentBlockIndex))
+        }
+        return .success(())
+    }
+    
+    var currentBlockIndex: UInt64 {
+        min(ledgersHighestKnownBlock, viewsHighestKnownBlock)
+    }
+    
+    private var viewLedgerOutOfSync: Bool {
+        abs(
```

Comment:
Since you're doing a max and min within this, do you really need the abs since you're guaranteed that the subtracted value will always be <= value it's subtracted from?

Reply:
same as above, avoids the possibility of a negative result in one of the subtraction operations, I will add a comment to disambiguate this for future readers.